Add additional math operations

ayselafsar opened the pull request:
- Get intersection of two 2d lines
- Get intersection of two rectangles
- Add sign operation in math module

ayselafsar committed:
Add additional math operations

- Get intersection of two 2d lines
- Get intersection of two rectangles
- Add sign operation in math module

swederik commented on src/lineSegment.js:
```diff
@@ -43,10 +43,95 @@ var cornerstoneMath = (function (cornerstoneMath) {
+            cornerstoneMath.sign(r1) === cornerstoneMath.sign(r2)) {
+            intersectionPoint.x = 0;
+            intersectionPoint.y = 0;
+            intersectionPoint.intersected = false;
```
Can you remove the 'intersected' attribute and just return undefined if there is no intersection?

This function should just return a point object with x and y coordinates, or undefined if no intersection exists.

ayselafsar replied:
This is done.

swederik commented on src/rect.js:
```diff
@@ -78,10 +78,110 @@ var cornerstoneMath = (function (cornerstoneMath) {
+        return intersectLeftRight && intersectTopBottom;
+    }
+
+    // Returns intersection points of two rectangles
```
Could you provide some further documentation?
- Does this only work for non-rotated rectangles?
- It appears that the function returns coordinates as left, right, top, and bottom. Could it be expressed as two points instead?

ayselafsar replied:
Yes, It only works for non-rotated rectangles.
It is updated to use two points.

swederik commented:
Hi Aysel,

I've made some comments inline which you can see above. Let me know if you need any clarification.

Thanks for the contribution!

ayselafsar committed:
Update additional math operations

- Unnecessary intersected property is removed
- Use top-left and bottom-right points  for rectangle
